python/sgl_jax/srt/models/mimo_v2_flash.py

```
@@ -0,0 +1,1017 @@
+            # 2D block-quant scale: (out_blocks, in_blocks) in HF layout.
+            # Expand to 3D kernel-ready (in_blocks, 1, out_dim) then reuse _block_dequant.
+            from sgl_jax.srt.kernels.quantized_matmul.blockwise_utils import (
+                expand_block_scale,
+            )
+
+            # Determine out_dim: weight_q is (in, out) model layout or (out, in) HF layout.
+            _, in_blocks = weight_scale.shape
+            if weight_q.shape[0] % in_blocks == 0:
+                out_dim = weight_q.shape[1]  # model layout (in, out)
+            else:
+                out_dim = weight_q.shape[0]  # HF layout (out, in)
```

Ruff (0.15.10)

[warning] Use ternary operator `out_dim = weight_q.shape[1] if weight_q.shape[0] % in_blocks == 0 else weight_q.shape[0]` instead of `if`-`else`-block

(SIM108)